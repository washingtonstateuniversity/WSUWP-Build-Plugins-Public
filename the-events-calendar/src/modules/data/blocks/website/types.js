/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/WEBSITE_BLOCK/SET_INITIAL_STATE`;

export const SET_WEBSITE_URL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_URL`;

export const SET_WEBSITE_LABEL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_LABEL`;
