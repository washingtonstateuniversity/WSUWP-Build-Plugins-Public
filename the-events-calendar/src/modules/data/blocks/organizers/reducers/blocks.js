/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
