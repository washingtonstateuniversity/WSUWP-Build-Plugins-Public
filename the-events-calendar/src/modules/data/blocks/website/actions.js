/**
 * Internal dependencies
 */
import * as types from './types';

export const setWebsite = ( url ) => ( {
	type: types.SET_WEBSITE_URL,
	payload: {
		url,
	},
} );

export const setLabel = ( label ) => ( {
	type: types.SET_WEBSITE_LABEL,
	payload: {
		label,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );
