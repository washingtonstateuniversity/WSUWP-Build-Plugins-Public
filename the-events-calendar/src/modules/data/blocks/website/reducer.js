/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	url: undefined,
	label: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_WEBSITE_URL:
			return {
				...state,
				url: action.payload.url,
			};
		case types.SET_WEBSITE_LABEL:
			return {
				...state,
				label: action.payload.label,
			};
		default:
			return state;
	}
};
