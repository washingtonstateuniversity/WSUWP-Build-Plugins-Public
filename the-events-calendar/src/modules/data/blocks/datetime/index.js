/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
