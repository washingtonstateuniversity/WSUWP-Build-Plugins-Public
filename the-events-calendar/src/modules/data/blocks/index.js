/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
