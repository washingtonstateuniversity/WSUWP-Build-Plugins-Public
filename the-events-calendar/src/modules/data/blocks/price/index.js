/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import sagas from './sagas';

export default reducer;
export { selectors, actions, types, sagas };
