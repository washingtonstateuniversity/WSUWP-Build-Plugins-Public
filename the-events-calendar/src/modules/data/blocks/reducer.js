/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime from './datetime';
import organizers from './organizers';
import price from './price';
import website from './website';
import venue from './venue';
import classic from './classic';
import sharing from './sharing';

export default combineReducers( {
	datetime,
	classic,
	venue,
	organizers,
	price,
	website,
	sharing,
} );
