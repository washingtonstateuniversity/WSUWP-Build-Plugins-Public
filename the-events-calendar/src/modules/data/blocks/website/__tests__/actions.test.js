/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/website';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Website actions', () => {
	it( 'Should set initial state', () => {
		expect( actions.setInitialState( {} ) ).toMatchSnapshot();
	} );

	it( 'Should set the website URL', () => {
		expect( actions.setWebsite( 'https://tri.be/' ) ).toMatchSnapshot();
	} );

	it( 'Should set the website label', () => {
		expect( actions.setLabel( 'Modern Tribe' ) ).toMatchSnapshot();
	} );
} );
