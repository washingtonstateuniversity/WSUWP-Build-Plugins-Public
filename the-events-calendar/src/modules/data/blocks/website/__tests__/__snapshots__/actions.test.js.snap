// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Website actions Should set initial state 1`] = `
Object {
  "payload": Object {},
  "type": "@@MT/EVENTS/WEBSITE_BLOCK/SET_INITIAL_STATE",
}
`;

exports[`[STORE] - Website actions Should set the website URL 1`] = `
Object {
  "payload": Object {
    "url": "https://tri.be/",
  },
  "type": "@@MT/EVENTS/SET_WEBSITE_URL",
}
`;

exports[`[STORE] - Website actions Should set the website label 1`] = `
Object {
  "payload": Object {
    "label": "Modern Tribe",
  },
  "type": "@@MT/EVENTS/SET_WEBSITE_LABEL",
}
`;
